include/hc_am.hpp

```diff
@@ -174,6 +174,28 @@ void am_memtracker_update_peers(const hc::accelerator &acc, int peerCnt, hsa_age
  */
 am_status_t am_map_to_peers(void* ptr, size_t num_peer, const hc::accelerator* peers); 
 
+/*
+ * Locks a host pointer to a vector of agents
+ * 
+ * @p ac acclerator corresponding to current device
+ * @p hostPtr pointer to host memory which should be page-locked
+ * @p size size of hostPtr to be page-locked
+ * @p accVec vector to hsa agents to which the hostPtr should be visible
+ * @return AM_SUCCESS if lock is successfully.
+ * @return AM_ERROR_MISC if lock is unsuccessful.
+ */
+am_status_t am_memory_host_lock(hc::accelerator &ac, void *hostPtr, size_t size, std::vector<hsa_agent_s> &accVec);
```

Comment:
could you consider remove the dependency of hsa_agent_s here? for example, change them to std::vector<void*> ? normally we want to avoid having direct HSA dependency in HCC public headers, so it's possible to build x86 simulation path or OpenCL path.